terraform/ec2/main.tf

```diff
@@ -405,9 +405,11 @@ resource "null_resource" "setup_sample_app_and_mock_server" {
   provisioner "remote-exec" {
     inline = [
       "sudo yum update -y",
-      "sudo amazon-linux-extras install docker -y",
+      "sudo yum install -y libxcrypt-compat",
+      "sudo yum install -y docker",
       "sudo service docker start",
       "sudo usermod -a -G docker ec2-user",
+      "sudo ln -s /usr/lib/libcrypt.so /usr/lib/libcrypt.so.1",
```

Comment:
Do we need this anymore since we are not updating sample app to al2023?

Reply:
We dont need them but it does not change any behavior on Al2022. Skipping from using `amazon-linux-extras` will make sure the future engineer not face the same issue when they try to upgrade the sample app instances to AL2023

Reply:
This didn't even work though right? There were additional issues on top? I would prefer to just remove this if it's not necessary. You can leave it as a comment if you'd like.

Reply:
They worked but there were additional issues relating to docker login. Okay, I have reverted the changes.